src/modules/keyboard/keyboardwidget/keyboardglobal.cpp

```diff
@@ -199,6 +199,48 @@ parseKeyboardLayouts( const char* filepath )
     return layouts;
 }
 
+static KeyboardGlobal::GroupsMap
+parseKeyboardGroupsSwitchers( const char* filepath )
+{
+    KeyboardGlobal::GroupsMap models;
+
+    QFile fh( filepath );
+    fh.open( QIODevice::ReadOnly );
+
+    if ( !fh.isOpen() )
+    {
+        cDebug() << "X11 Keyboard model definitions not found!";
+        return models;
+    }
+
+    bool modelsFound = findSection( fh, "! option" );
+    // read the file until the end or until we break the loop
+    while ( modelsFound && !fh.atEnd() )
+    {
+        QByteArray line = fh.readLine();
+
+        // check if we start a new section
+        if ( line.startsWith( '!' ) )
+        {
+            break;
+        }
+
+        // here we are in the model section, otherwise we would continue or break
+        QRegExp rx;
+        rx.setPattern( "^\\s+grp:(\\S+)\\s+(\\w.*)\n$" );
```

Comment:
I just ported all the modules to be Qt6 compatible, which means away from `QRegExp` and to `QRegularExpression`. Take a look at 427311f2c31600e87879ddb1587d5a65f30e3645 for how it can be done.